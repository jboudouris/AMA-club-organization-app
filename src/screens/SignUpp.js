import React from 'react'
import { Image, TouchableOpacity, ImageBackground, StyleSheet, Text, TextInput, View, Button, SafeAreaView } from 'react-native'
import firebase from 'firebase';
import { db } from '../config';
let config = {
  apiKey: "AIzaSyB9kMpxhilLczlWR57TLOM8IMxKuUEcRW0",
    authDomain: "react-firebase-cd68d.firebaseapp.com",
    databaseURL: "https://react-firebase-cd68d.firebaseio.com",
    projectId: "react-firebase-cd68d",
    storageBucket: "react-firebase-cd68d.appspot.com",
    messagingSenderId: "57969678002"
};

export default class SignUp extends React.Component {
  state = {
    email: '',
    password: '',
    confirmedPassword: '',
    first_Name: '',
    last_Name: '',
    req: '',
    errorMessage: null
  }
  handleSignUp = () => {
      firebase
        .auth()
        .createUserWithEmailAndPassword(this.state.email, this.state.password)
        .then(
          (user)=>{
            if(user){
              if(this.state.email.includes("@wisc.edu")) {
                db.ref('/user/' + user.uid).update({
                    email: this.state.email,
                    password: this.state.password,
                    userKey: user.uid,
                    first_Name: this.state.first_Name,
                    last_Name: this.state.last_Name,
                    attendantNum: 0,
                    role: 'user',
                    quote: '',
                    phone_Number: '',
                    alt_Email: '',
                });
              this.sendVerification();
              firebase.auth().signOut();

              //firebase.auth().signOut();
              //alert(user.uid);
              }
            }
          }
        )
        .then(() => this.props.navigation.navigate('Loginn'))
        .catch(error => this.setState({ errorMessage: error.message }))

    }

  sendVerification = () => {
    let user = firebase.auth().currentUser;
    user.sendEmailVerification().then(function() {

    }).catch(function(error) {

    });
  }

  checkEmail = () => {
    if (this.state.email.includes("@wisc.edu") == true){
      if (this.state.password == this.state.confirmedPassword)
      {
        this.handleSignUp();
      }
      else
      {
        alert('Passwords do not match.');
      }
    }
   else {
      alert("Please enter your Wisc email (@wisc.edu).");
    }
  }

  handleEmailChange = (email) => {
    this.setState({
      email: email
    });
  }

  handlePasswordChange = (password) => {
    this.setState({
      password: password
    });
  }

  handleConPasswordChange = (confirmedPassword) => {
    this.setState({
      confirmedPassword: confirmedPassword
    });
  }
  handleFirstNameChange = (first_Name) => {
    this.setState({
      first_Name: first_Name
    });
  }
  handleLastNameChange = (last_Name) => {
    this.setState({
      last_Name: last_Name
    });
  }
render() {
    return (
      <SafeAreaView style={styles.safeArea}>
      <View style={styles.container}>
          <ImageBackground
              style = {styles.backgroundImage}
              source = {require('../backgrounds/BG2.png')}
          >
             <View style={styles.container}>
                <Image
                    style = {{alignSelf: 'center', width: 150, height: 50, margin: 10}}
                    source = {require('../icons/AMA_white.png')}
                />
                {this.state.errorMessage &&
                  <Text style={{ color: 'red' }}>
                    {this.state.errorMessage}
                  </Text>}
                <TextInput
                  placeholder="First Name"
                  autoCapitalize="none"
                  placeholderTextColor= 'white'
                  style={styles.textInput}
                  onChangeText={first_Name => this.handleFirstNameChange( first_Name )}
                  value={this.state.first_Name}
                />
                <TextInput
                  placeholder="Last Name"
                  autoCapitalize="none"
                  placeholderTextColor= 'white'
                  style={styles.textInput}
                  onChangeText={last_Name => this.handleLastNameChange( last_Name )}
                  value={this.state.last_Name}
                />
                <TextInput
                  placeholder="Email"
                  autoCapitalize="none"
                  placeholderTextColor= 'white'
                  style={styles.textInput}
                  onChangeText={email => this.handleEmailChange( email )}
                  value={this.state.email}
                />
                <TextInput
                  secureTextEntry
                  placeholder="Password"
                  autoCapitalize="none"
                  placeholderTextColor= 'white'
                  style={styles.textInput}
                  onChangeText={password => this.handlePasswordChange( password )}
                  value={this.state.password}
                />
                <TextInput
                  secureTextEntry
                  placeholder="Confirm Password"
                  autoCapitalize="none"
                  placeholderTextColor= 'white'
                  style={styles.textInput}
                  onChangeText={confirmedPassword => this.handleConPasswordChange( confirmedPassword )}
                  value={this.state.confirmedPassword}
                />
                <TouchableOpacity style = {styles.btn1} onPress={this.checkEmail}>
                    <Text style={styles.btntxt}>Sign Up</Text>
                </TouchableOpacity>
                <TouchableOpacity style = {styles.btn1} onPress={() => this.props.navigation.navigate('Loginn')}>
                    <Text style={styles.btntxt}>Already have an account? Log in here</Text>
                </TouchableOpacity>
            </View>
        </ImageBackground>
      </View>
      </SafeAreaView>
    )
  }
}

const styles = StyleSheet.create({
    backgroundImage: {
        flex: 1,
        alignSelf: 'stretch',
    },
    btn1: {
        backgroundColor: 'rgba(127,141,221,0.6)',
        alignSelf: 'center',
        width: '90%',

        borderWidth: 1,
        marginTop: 10,
        borderColor: 'white',
        height: 50,
        alignItems: 'center',
        justifyContent: 'center',
    },
    btntxt: {
        fontSize: 18,
        color: 'white',
    },
    buttonView: {
        flexDirection:'row',
        backgroundColor: 'transparent',
    },
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  textInput: {
    height: 50,
    fontSize: 18,
    width: '90%',
    borderColor: 'white',
    backgroundColor: 'rgba(57,189,225,0.6)',
    borderWidth: 1,
    marginTop: 8,
    color: 'white',
  },
  safeArea: {
  flex: 1,
  backgroundColor: '#ddd'
}
})
