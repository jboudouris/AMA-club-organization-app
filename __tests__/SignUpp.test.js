import React from 'react';
import renderer from 'react-test-renderer';
import SignUpp from '../src/screens/SignUpp';

describe('Testing SignUpp', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <SignUpp />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

it('should change the email value', () => {
    let loginComponent = renderer.create(<SignUpp/>).getInstance()

    loginComponent.handleEmailChange('some_email')

    expect(loginComponent.state.email).toEqual('some_email')
})

it('should change the password value', () => {
    let loginComponent = renderer.create(<SignUpp/>).getInstance()

    loginComponent.handlePasswordChange('some_password')

    expect(loginComponent.state.password).toEqual('some_password')
})


it('should change the confirmed password value', () => {
    let loginComponent = renderer.create(<SignUpp/>).getInstance()

    loginComponent.handleConPasswordChange('some_password')

    expect(loginComponent.state.confirmedPassword).toEqual('some_password')
})

it('should change the first name value', () => {
    let loginComponent = renderer.create(<SignUpp/>).getInstance()

    loginComponent.handleFirstNameChange('some_name')

    expect(loginComponent.state.first_Name).toEqual('some_name')
})

it('should change the last name value', () => {
    let loginComponent = renderer.create(<SignUpp/>).getInstance()

    loginComponent.handleLastNameChange('some_last')

    expect(loginComponent.state.last_Name).toEqual('some_last')
})

it('should check email', () => {
    let loginComponent = renderer.create(<SignUpp/>).getInstance()
    loginComponent.handleEmailChange('some_email')
    loginComponent.checkEmail()

    expect(loginComponent.checkEmail()).toBeTruthy()
})
