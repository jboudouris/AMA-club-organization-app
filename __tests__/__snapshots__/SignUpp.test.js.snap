// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing SignUpp renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#ddd",
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "alignSelf": "stretch",
          "flex": 1,
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../../src/backgrounds/BG2.png",
          }
        }
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": undefined,
              "width": undefined,
            },
            undefined,
          ]
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "flex": 1,
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../src/icons/AMA_white.png",
            }
          }
          style={
            Object {
              "alignSelf": "center",
              "height": 50,
              "margin": 10,
              "width": 150,
            }
          }
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="First Name"
          placeholderTextColor="white"
          style={
            Object {
              "backgroundColor": "rgba(57,189,225,0.6)",
              "borderColor": "white",
              "borderWidth": 1,
              "color": "white",
              "fontSize": 18,
              "height": 50,
              "marginTop": 8,
              "width": "90%",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Last Name"
          placeholderTextColor="white"
          style={
            Object {
              "backgroundColor": "rgba(57,189,225,0.6)",
              "borderColor": "white",
              "borderWidth": 1,
              "color": "white",
              "fontSize": 18,
              "height": 50,
              "marginTop": 8,
              "width": "90%",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Email"
          placeholderTextColor="white"
          style={
            Object {
              "backgroundColor": "rgba(57,189,225,0.6)",
              "borderColor": "white",
              "borderWidth": 1,
              "color": "white",
              "fontSize": 18,
              "height": 50,
              "marginTop": 8,
              "width": "90%",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Password"
          placeholderTextColor="white"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(57,189,225,0.6)",
              "borderColor": "white",
              "borderWidth": 1,
              "color": "white",
              "fontSize": 18,
              "height": 50,
              "marginTop": 8,
              "width": "90%",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <TextInput
          allowFontScaling={true}
          autoCapitalize="none"
          onChangeText={[Function]}
          placeholder="Confirm Password"
          placeholderTextColor="white"
          secureTextEntry={true}
          style={
            Object {
              "backgroundColor": "rgba(57,189,225,0.6)",
              "borderColor": "white",
              "borderWidth": 1,
              "color": "white",
              "fontSize": 18,
              "height": 50,
              "marginTop": 8,
              "width": "90%",
            }
          }
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "rgba(127,141,221,0.6)",
              "borderColor": "white",
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 10,
              "opacity": 1,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
              }
            }
          >
            Sign Up
          </Text>
        </View>
        <View
          accessible={true}
          isTVSelectable={true}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "center",
              "backgroundColor": "rgba(127,141,221,0.6)",
              "borderColor": "white",
              "borderWidth": 1,
              "height": 50,
              "justifyContent": "center",
              "marginTop": 10,
              "opacity": 1,
              "width": "90%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontSize": 18,
              }
            }
          >
            Already have an account? Log in here
          </Text>
        </View>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
